cmd/mvalidate/main.go

```diff
@@ -0,0 +1,121 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	flag.Parse()
+	yml := flag.Arg(0)
+	if err := run(yml); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(ymlPath string) error {
+	if ymlPath == "" {
+		return errors.New("argument must be metadata.yaml file")
+	}
+	ymlDir, err := filepath.Abs(ymlPath)
+	if err != nil {
+		return fmt.Errorf("cannot compute the absolute path of %s", ymlDir)
+	}
+	ymlDir = path.Dir(ymlDir)
+	if _, err := os.Stat(ymlPath); os.IsNotExist(err) {
+		log.Printf("metadata.yaml missing for %s, skipping\n", ymlDir)
+		return nil
+	}
+
+	md, err := loadMetadata(filepath.Clean(ymlPath))
+	if err != nil {
+		return fmt.Errorf("failed loading %v: %w", ymlPath, err)
+	}
+	if md.Status == nil {
+		return errors.New("missing status metadata")
+	}
+	if md.Type == "" {
+		return errors.New("missing type metadata")
+	}
+	if md.Status.Class == "" {
+		return errors.New("missing status class metadata")
+	}
+	if md.Status.Stability == "" {
+		return errors.New("missing status stability metadata")
+	}
```

Comment:
I think these checks should be part of `metadata.Validate()` method in the mdatagen module

Reply:
That's good I will move that.

Reply:
Done here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21274